task_manager/task_manager.hh

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "replica/database.hh"
+
+class task_manager {
+public:
+    static const size_t modules_count = 2;
+    enum class module {
+        repair = 0,
+        compaction
+    };
+    class running_tasks {
+        std::vector<uint64_t> number;
+
+    public:
+        running_tasks() : number(modules_count) {}
+
+        uint64_t get_number(module m) {
+            return number[static_cast<int>(m)];
+        }
+
+        void add_task(module m) {
+            ++number[static_cast<int>(m)];
+        }
+
+        void delete_task(module m) {
+            --number[static_cast<int>(m)];
+        }
+
+        running_tasks& operator+=(const running_tasks& rhs) {
+            for (int i = 0; i < modules_count; ++i) {
+                number[i] += rhs.number[i];
+            }
+            return *this;
+        }
+ 
+        friend running_tasks operator+(const running_tasks& lhs, const running_tasks& rhs) {
+            running_tasks rt{lhs};
+            rt += rhs;
+            return rt;
+        }
+    };
+    static module get_module(std::string module) {
+        if (module == "repair") {
+            return module::repair;
+        } else {
+            return module::compaction;
+        }
+    }
+    enum class task_state {
+        running,
+        blocked,
+        paused,
```

Comment:
not sure we can use it.
better to add the states when they become used rather than having unused states